TicTacToeClassLibrary/TicTacToe.cs

```diff
@@ -1,38 +1,47 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-namespace TicTacToeClassLibrary
+﻿namespace TicTacToeClassLibrary
 {
     public class TicTacToe
     {
-        private List<IMemento> _mementos;
-        private IBoardFactory _boardFactory;
+        private readonly List<IMemento> _mementos;
+        private readonly IBoardFactory _boardFactory;
+
         public Board TicTacToeBoard { get; set; }
         public Player Player1 { get; set; }
         public Player Player2 { get; set; }
         public Player CurrentPlayer { get; set; }
+
         public TicTacToe()
         {
             _boardFactory = new BoardFactory();
-            TicTacToeBoard = Create();
-            Console.WriteLine("Enter the name of the first player: ");
-            Player1 = new Player(Console.ReadLine(), "x", 0);
-            Console.WriteLine("Enter the name of the second player: ");
-            Player2 = new Player(Console.ReadLine(), "o", 0);
+            _mementos = new List<IMemento>();
+
+            TicTacToeBoard = CreateBoard();
+            InitializePlayers();
             CurrentPlayer = Player1;
             PrintGameInfo();
         }
-        public void SwitchCurrentPlayer()
+
+        private Board CreateBoard()
         {
-            CurrentPlayer = CurrentPlayer == Player1 ? Player2 : Player1;
+            Console.WriteLine("Please select a board type (3x3, 4x4)");
+            string boardType = Console.ReadLine();
+            return _boardFactory.Create(boardType);
+        }
```

Comment:
In my opinion, there is no need to create two separate CreateBoard() and RestartGame() methods because they duplicate the already written Create() method.